framework/src/utils/Conversion.C

```diff
@@ -212,20 +215,31 @@ namespace Moose
   {
     switch (t)
     {
-    case EXEC_INITIAL:        return "INITIAL";
-    case EXEC_LINEAR:         return "LINEAR";
-    case EXEC_NONLINEAR:      return "NONLINEAR";
-    case EXEC_TIMESTEP_END:   return "TIMESTEP_END";
-    case EXEC_TIMESTEP_BEGIN: return "TIMESTEP_BEGIN";
-    case EXEC_CUSTOM:         return "CUSTOM";
-    case EXEC_FINAL:          return "FINAL";
-    case EXEC_FORCED:         return "FORCED";
-    case EXEC_FAILED:         return "FAILED";
-    case EXEC_SUBDOMAIN:      return "SUBDOMAIN";
-    case EXEC_NONE:           return "NONE";
+      case EXEC_INITIAL:        return "INITIAL";
+      case EXEC_LINEAR:         return "LINEAR";
+      case EXEC_NONLINEAR:      return "NONLINEAR";
+      case EXEC_TIMESTEP_END:   return "TIMESTEP_END";
+      case EXEC_TIMESTEP_BEGIN: return "TIMESTEP_BEGIN";
+      case EXEC_CUSTOM:         return "CUSTOM";
+      case EXEC_FINAL:          return "FINAL";
+      case EXEC_FORCED:         return "FORCED";
+      case EXEC_FAILED:         return "FAILED";
+      case EXEC_SUBDOMAIN:      return "SUBDOMAIN";
+      case EXEC_NONE:           return "NONE";
     }
     return "";
   }
+
+  template<>
+  std::string stringify(const Real & t)
+  {
+    std::ostringstream os;
+    // the correct value would be std::numeric_limits<Real>::max_digits10 (C++11)
+    os << std::scientific
+       << std::setprecision(std::numeric_limits<Real>::digits10 + 1)
```

Comment:
Ok, I'm going with http://en.cppreference.com/w/cpp/types/numeric_limits/max_digits10 and implement `Moose::numeric_limits<T>::max_digits10`